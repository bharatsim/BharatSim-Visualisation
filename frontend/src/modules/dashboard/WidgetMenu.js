import React from 'react';
import PropTypes from 'prop-types';
import { IconButton, Typography, useTheme } from '@material-ui/core';
import { DeleteOutline, MoreVert } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';
import DeleteConfirmationModal from '../../uiComponent/DeleteConfirmationModal';
import useModal from '../../hook/useModal';
import DropdownMenu from '../../uiComponent/DropdownMenu/DropdownMenu';

const iconStyles = makeStyles((theme) => ({
  root: {
    padding: 0,
    height: theme.spacing(5),
    width: theme.spacing(5),
  },
}));
export default function WidgetMenu({ onDelete }) {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const theme = useTheme();
  const iconClasses = iconStyles();
  const {
    isOpen: isDeleteModalOpen,
    openModal: openDeleteModal,
    closeModal: closeDeleteModal,
  } = useModal();

  const openMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };
  const closeMenu = () => {
    setAnchorEl(null);
  };

  function onOpenDeleteModal() {
    closeMenu();
    openDeleteModal();
  }

  function handleDelete() {
    closeDeleteModal();
    onDelete();
  }

  return (
    <>
      <IconButton data-testid="widget-menu" disableRipple onClick={openMenu} classes={iconClasses}>
        <MoreVert htmlColor={theme.colors.grayScale['500']} />
      </IconButton>
      <DropdownMenu
        anchorEl={anchorEl}
        closeMenu={closeMenu}
        menuItems={[
          {
            label: 'Delete Chart',
            dataTestId: 'DeleteChart',
            onClick: onOpenDeleteModal,
            icon: <DeleteOutline />,
          },
        ]}
      />
      {isDeleteModalOpen && (
        <DeleteConfirmationModal
          handleClose={closeDeleteModal}
          open={isDeleteModalOpen}
          title="Delete Chart"
          deleteAction={{
            onDelete: handleDelete,
            name: 'Delete Chart',
            dataTestId: 'delete-chart-confirm',
          }}
        >
          <Typography variant="body2">
            Are you sure you want to delete this chart and all it’s configurations? This action
            can’t be undone
          </Typography>
        </DeleteConfirmationModal>
      )}
    </>
  );
}

WidgetMenu.propTypes = {
  onDelete: PropTypes.func.isRequired,
};
