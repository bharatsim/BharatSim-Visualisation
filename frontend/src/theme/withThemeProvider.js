import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
