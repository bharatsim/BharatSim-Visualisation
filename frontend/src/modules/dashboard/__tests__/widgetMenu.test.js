import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import WidgetMenu from '../WidgetMenu';
import withThemeProvider from '../../../theme/withThemeProvider';

describe('<widgetMenu>', () => {
  it('should render all the menu', () => {
    const MenuWithProviders = withThemeProvider(WidgetMenu);
    const { getByTestId } = render(<MenuWithProviders onDelete={() => {}} onEdit={() => {}} />);
    fireEvent.click(getByTestId('widget-menu'));

    expect(document.body).toMatchSnapshot();
  });
});
