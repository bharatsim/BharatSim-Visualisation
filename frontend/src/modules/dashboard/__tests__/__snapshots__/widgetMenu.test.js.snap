// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<widgetMenu> should render all the menu 1`] = `
<div
  aria-hidden="true"
>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    data-testid="widget-menu"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </span>
  </button>
</div>
`;
